### ISSUE TITLE
TODO: Implement Trajectory Imitation Loss

### ISSUE THREAD

**Issue by riita10069** [autowarefoundation/auto_e2e#15]
## Overview

DrivingPolicy outputs a 128-dim trajectory prediction but currently has no loss to train against. This issue tracks what is needed to define the trajectory loss.

## Items to consider

- [ ] Loss function type (Smooth L1 / MSE / other)
- [ ] Trajectory format: 64 timesteps × (acceleration, curvature) at 10Hz = 6.4s
- [ ] Whether temporal weighting is needed (e.g., weight near-future more heavily)
- [ ] Additional penalty terms (collision, lane departure) — possibly later

## Notes

This is the primary task loss for the model. The specific design will likely depend on which dataset we use and what format the ground truth trajectories come in. Would love to hear the team's thoughts on what has worked well in similar setups.

## References

- UniAD uses L1 trajectory loss + collision penalty
- VAD uses vectorized trajectory representation

**Comment by riita10069** [comment]
Just a note as I was reviewing the current state of things — since the forward pass and tests are all working now, the next step to actually make the model learn would be defining the trajectory loss. Without it, the DrivingPolicy parameters stay at random initialization even though the architecture is in place.

I think this is also upstream of a few other items — the feature reconstruction loss (#13), training loop (#14), and BEV segmentation (#17) all depend on having at least the primary loss defined first.

A few questions for the team:

- Is there a preferred ground truth format? The current output is 64 × (acceleration, curvature) at 10Hz, but if the dataset provides (x, y) waypoints we'd need to adjust the output head.
- Would it make sense to start with a simple MSE on synthetic trajectories just to validate the training loop end-to-end, before committing to a final loss design?

Happy to help with implementation once there's alignment on direction. No rush — just wanted to surface this as a dependency.

**Comment by m-zain-khawaja** [comment]
Thanks @riita10069 

- I think a synthetic trajectory to validate training loop would be useful
- You are right that datasets typically provide trajectory ground truth as a waypoints list, however, I think the (acceleration, curvature) prediction was found to be more stable by Nvidia in their Alpayamo paper, below are excerpts:

<img width="615" height="297" alt="Image" src="https://github.com/user-attachments/assets/cfaeea08-3a43-4f4e-b974-978f4e53bcd1" />

<img width="605" height="244" alt="Image" src="https://github.com/user-attachments/assets/4be2b121-f1f9-4cf0-8dc3-61b46bc56167" />


References:
https://arxiv.org/abs/2511.00088

**Comment by m-zain-khawaja** [comment]
Thanks @riita10069 - PR is merged
